options.go

```diff
@@ -177,8 +197,38 @@ func IdentityResolverFunc(resolver IdentityResolver) Option {
 	}
 }
 
+// RingChecksumStatPeriod configures the period between emissions of the stat
+// 'ring.checksum-periodic'. Using a value <=0 (or RingChecksumStatPeriodNever)
+// will disable emission of this stat. Using a value in (0, 10ms) will return
+// an error, as that value is unrealistically small. Normal values must
+// therefore be >=10ms. RingChecksumStatPeriodDefault defines the default.
+func RingChecksumStatPeriod(period time.Duration) Option {
+	return func(r *Ringpop) error {
+		if period <= 0 {
+			period = RingChecksumStatPeriodNever
+		} else if period < 10*time.Millisecond {
+			return errors.New("RingChecksumStatPeriod invalid below 10 ms")
+		}
+		r.config.RingChecksumStatPeriod = period
+		return nil
+	}
+}
+
 // Default options
 
+// RingChecksumStatPeriodNever defines a "period" which disables
+// ring.checksum-periodic stat emission.
+const RingChecksumStatPeriodNever = time.Duration(-1)
```

Comment:
I think it's okay to make `0` mean "disabled", as long as it's clear in the doc comment. I can see the reason for the constant, but I think it actually makes it more complicated than it needs to be.

This is what disabling the checksum would look like with the constant:

```go
rp := ringpop.New("myapp",
    ringpop.Channel(ch),
    ringpop.RingChecksumStatPeriod(ringpop.RingChecksumStatPeriodNever))
```

Without it:

```go
rp := ringpop.New("myapp",
    ringpop.Channel(ch),
    ringpop.RingChecksumStatPeriod(0))   // If you really really need to do this, you can
```

As a side note, I don't see a strong use-case for ever disabling this anyway.